### PR TITLE
added regression test

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/7913

### CODE REVIEW

File: tests/PHPStan/Rules/Arrays/data/bug-7913.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types = 1);
+
+namespace Bug7913;
+
+const X = [];
```
Comment (staabm) [inline tests/PHPStan/Rules/Arrays/data/bug-7913.php:5]: I tried wrapping it in a method/function, but using `const` like in the repro example, only works in top level code, it seems

**Comment by ondrejmirtes** [comment]
I'd like a NodeScopeResolverTest with assertType before and after empty().

**Comment by ondrejmirtes** [comment]
Rebase please :) (always make sure you're on the latest commit before starting the work).

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**